libkbfs/tlf_journal.go

```diff
@@ -1379,17 +1447,39 @@ func (j *tlfJournal) getBlockData(id kbfsblock.ID) (
 func (j *tlfJournal) putBlockData(
 	ctx context.Context, id kbfsblock.ID, context kbfsblock.Context, buf []byte,
 	serverHalf kbfscrypto.BlockCryptKeyServerHalf) error {
+	// Since Acquire can block, it should happen outside of the
+	// journal lock.
+	bufLen := int64(len(buf))
+	j.log.CDebugf(ctx, "Acquiring %d bytes for %s", bufLen, j.tlfID)
+	availableBytes, err := j.diskLimiter.Acquire(ctx, bufLen)
```

Comment:
Done, added separate error with a shorter (3s) timeout, and test. Will file bug for nice UI error for disk full.